cg/services/file_delivery/deliver_files_service/deliver_files_service_factory.py

```diff
@@ -0,0 +1,106 @@
+    SampleFileFormatter,
+)
+from cg.services.file_delivery.move_files_service.move_delivery_files_service import (
+    MoveDeliveryFilesService,
+)
+from cg.store.store import Store
+
+
+class DeliveryServiceFactory:
+    """Class to build the delivery services based on workflow and delivery type."""
+
+    def __init__(self, store: Store, hk_api: HousekeeperAPI):
+        self.store = store
+        self.hk_api = hk_api
+
+    @staticmethod
+    def _get_file_tag_fetcher(delivery_type: DataDelivery) -> FetchDeliveryFileTagsService:
+        """Get the file tag fetcher based on the delivery type."""
+        service_map: dict[DataDelivery, FetchDeliveryFileTagsService] = {
+            DataDelivery.FASTQ: FetchSampleAndCaseDeliveryFileTagsService,
+            DataDelivery.ANALYSIS_FILES: FetchSampleAndCaseDeliveryFileTagsService,
+            DataDelivery.FASTQ_ANALYSIS: FetchSampleAndCaseDeliveryFileTagsService,
+        }
+        return service_map[delivery_type]()
+
+    def _get_file_fetcher(self, delivery_type: DataDelivery) -> FetchDeliveryFilesService:
+        """Get the file fetcher based on the delivery type."""
+        service_map: dict[DataDelivery, FetchDeliveryFilesService] = {
+            DataDelivery.FASTQ: FetchFastqDeliveryFilesService,
+            DataDelivery.ANALYSIS_FILES: FetchAnalysisDeliveryFilesService,
+            DataDelivery.FASTQ_ANALYSIS: FetchFastqAndAnalysisDeliveryFilesService,
+        }
+        file_tag_fetcher: FetchDeliveryFileTagsService = self._get_file_tag_fetcher(delivery_type)
+        return service_map[delivery_type](
+            status_db=self.store,
+            hk_api=self.hk_api,
+            tags_fetcher=file_tag_fetcher,
+        )
+
+    @staticmethod
+    def _get_sample_file_formatter(
+        workflow: Workflow,
+    ) -> SampleFileFormatter | SampleFileConcatenationFormatter:
+        """Get the file formatter service based on the workflow."""
+        if workflow in [Workflow.MICROSALT, Workflow.MUTANT]:
+            return SampleFileConcatenationFormatter(FastqConcatenationService())
+        return SampleFileFormatter()
+
+    def build_delivery_service(
+        self, workflow: Workflow, delivery_type: DataDelivery
+    ) -> DeliverFilesService:
+        """Build a delivery service based on the workflow and delivery type."""
+        file_fetcher: FetchDeliveryFilesService = self._get_file_fetcher(delivery_type)
+        sample_file_formatter: SampleFileFormatter | SampleFileConcatenationFormatter = (
+            self._get_sample_file_formatter(workflow)
+        )
+        file_formatter: DeliveryFileFormattingService = DeliveryFileFormatter(
+            case_file_formatter=CaseFileFormatter(), sample_file_formatter=sample_file_formatter
+        )
+        return DeliverFilesService(
+            delivery_file_manager_service=file_fetcher,
+            move_file_service=MoveDeliveryFilesService(),
```

Comment:
Do we need to provide the move_file_service in the instantiation of the DeliverFilesService or will it always be a freshly created MoveDeliveryFilesService?

Reply:
yes, it is injected to the DeliveryService